.travis.yml

```diff
@@ -100,14 +100,104 @@ jobs:
             - libwww-perl
             - g++-5
             - libubsan0
-            - parallel
-            - libc6-dev-i386
+      # env: COMPILER=g++-5 SAN_FLAGS="-fsanitize=undefined -fno-sanitize-recover -fno-omit-frame-pointer"
+      env:
+        - COMPILER="ccache /usr/bin/g++-5"
+        - EXTRA_CXXFLAGS="-D_GLIBCXX_DEBUG"
+      script: echo "Tests are run in the next stage."
+
+    # cmake build using g++-5
+    - stage: Linter + Doxygen + non-debug builds
+      os: linux
+      compiler: gcc
+      cache: ccache
+      env:
+        - BUILD_SYSTEM=cmake
+      addons:
+        apt:
+          sources:
+            - ubuntu-toolchain-r-test
+          packages:
+            - g++-5
+      install:
+        - ccache -z
+        - ccache --max-size=1G
+        - cmake -H. -Bbuild '-DCMAKE_BUILD_TYPE=Release'  '-DCMAKE_CXX_COMPILER=/usr/bin/g++-5'
+        - cmake --build build -- -j4
+      script: echo "Tests are run in the next stage."
+
+    # OS X using clang++
+    - stage: Linter + Doxygen + non-debug builds
+      os: osx
+      sudo: false
+      compiler: clang
+      cache: ccache
       before_install:
-        - mkdir bin ; ln -s /usr/bin/gcc-5 bin/gcc
+        - HOMEBREW_NO_AUTO_UPDATE=1 brew install ccache parallel
+        - export PATH=$PATH:/usr/local/opt/ccache/libexec
+      env: COMPILER="ccache clang++"
+      script: echo "Tests are run in the next stage."
+
+    # Ubuntu Linux with glibc using clang++-3.7, no-debug mode
+    - stage: Linter + Doxygen + non-debug builds
+      os: linux
+      sudo: false
+      compiler: clang
+      cache: ccache
+      addons:
+        apt:
+          sources:
+            - ubuntu-toolchain-r-test
+            - llvm-toolchain-precise-3.7
+          packages:
+            - libwww-perl
+            - clang-3.7
+            - libstdc++-5-dev
+            - libubsan0
+      before_install:
+      # env: COMPILER=clang++-3.7 SAN_FLAGS="-fsanitize=undefined -fno-sanitize-recover=undefined,integer -fno-omit-frame-pointer"
+      env:
+        - COMPILER="ccache /usr/bin/clang++-3.7"
+        - EXTRA_CXXFLAGS="-Qunused-arguments -fcolor-diagnostics -DNDEBUG"
+        - CCACHE_CPP2=yes
+      script: echo "Tests are run in the next stage."
+
+    # Ubuntu Linux with glibc using g++-5
+    - stage: Linter + Doxygen + non-debug builds
+      os: linux
+      sudo: false
+      compiler: gcc
+      cache: ccache
+      addons:
+        apt:
+          sources:
+            - ubuntu-toolchain-r-test
+          packages:
+            - libwww-perl
+            - g++-5
+            - libubsan0
```

Comment:
Is libubsan0 actually needed for this build, is it a hangover from when we used to have sanitizer builds? (ditto for elsewhere in this file as well)